### ISSUE TITLE
test of `collect_n_subject.R`

### ISSUE THREAD

**Issue by LittleBeannie** [Merck/metalite#84]

**Comment by LittleBeannie** [comment]
Test plan of `meta_remove_blank_group`
- Throw error message `"meta_remove_blank_group: parameter is not available in meta$population"' when run `meta_remove_blank_group(meta_dummy(), population = "apat", parameter = "apat")`
- The subjects with missing grouping variables will be removed after run `meta_remove_blank_group`. But need to construct fake data.

**Comment by LittleBeannie** [comment]
Test plan of `collect_n_subject`

TBA

**Comment by wangben718** [comment]
Hi @elong0527 , when dealing with NA value in `n_subject()`, do we only consider `NA` as missing value not including `""` or `NULL`?

**Comment by elong0527** [comment]
“” is not missing value that should be handle outside of mentality.

Null should be handled. If it does not work, please create a ticket with reproducible example.

**Comment by wangben718** [comment]
Hi @elong0527 , I got an error when I try to run:
`meta <- meta_dummy()`
`meta_remove_blank_group(meta, "apat", "rel")`

The error message I got is: Error in table(is.na(pop[[pop_var]]), pop[[pop_grp]]) : 
  all arguments must have the same length

Could you help me with it?

**Comment by wangben718** [comment]
Hi @elong0527 , when I was working on the validation. I feel this function does not work as expected when `use_na = 'no'`

`suppressWarnings(
    meta <- metalite::meta_example() |>
      define_parameter(name = "age", var = "AGE", label = "Age")
  )
meta$data_population$AGE[2] <- NA
  meta$data_population$TRTA[1:3] <- NA
  test3 <- collect_n_subject(meta, "apat", "age", use_na = "no")`

**Comment by wangben718** [comment]
> Hi @elong0527 , when I was working on the validation. I feel this function does not work as expected when `use_na = 'no'`
> 
> ```
> suppressWarnings(
>     meta <- metalite::meta_example() |>
>       define_parameter(name = "age", var = "AGE", label = "Age")
>   )
> meta$data_population$AGE[2] <- NA
> meta$data_population$TRTA[1:3] <- NA
> collect_n_subject(meta, "apat", "age", use_na = "no")
> ```
> 
> I am expecting no `missing` on column or row. But this is the snapshot of output I got.
> 
> ![image](https://user-images.githubusercontent.com/85646030/226520576-ccf4abcf-630f-45e3-9752-75e6576fdb5d.png)

Now I feel we are always using 'ifany' in `collect_n_subject()` now since the argument value actually is not used in sub-function `n_subject`. See https://github.com/Merck/metalite/blob/7416833b64dcd5b51be859f8be6b3f2f093a18aa/R/collect_n_subject.R#L266

**Comment by wangben718** [comment]
Hi @elong0527.  

For test case:
```
suppressWarnings(
    meta <- meta_example() |>
      define_parameter(name = "age", var = "AGE", label = "Age") )
test1 <- collect_n_subject(meta, "apat", "age", use_na = "always")
```
pop_n:
![image](https://user-images.githubusercontent.com/85646030/236981558-b3ef952f-5803-44f5-b283-37a08eb243af.png)

pop_all:
![image](https://user-images.githubusercontent.com/85646030/236981618-cc4bd576-d644-47b5-8ba9-cd9b6d758eeb.png)

pop_num:
![image](https://user-images.githubusercontent.com/85646030/236981863-92096f24-8f4a-4705-9247-6e5fee4bc59c.png)

**Comment by wangben718** [comment]
For test case:
```
suppressWarnings(
    meta <- meta_example() |>
      define_parameter(name = "age", var = "AGE", label = "Age") )
meta$data_population$AGE[2] <- NA
meta$data_population$TRTA[1:3] <- NA
test3 <- collect_n_subject(meta, "apat", "age", use_na = "no")
```
![image](https://user-images.githubusercontent.com/85646030/236983281-2b34500e-3ef7-4718-b50b-305e308b1be0.png)

**Comment by fb-elong** [comment]
Thanks, Could you also add them into testthat folder? So we know if it passed the CI/CD.

**Comment by wangben718** [comment]
Hi @elong0527 . Is it designed to always have a row with NA value in subset crieteria?
<img width="747" alt="image" src="https://github.com/Merck/metalite/assets/85646030/a2e076db-6886-407e-9d92-e7ff12ac65fc">

**Comment by fb-elong** [comment]
Yes, it will be useful if people needs to find those subjects later.
